### ISSUE TITLE
🐛 `linter.rules.recommended: false` doesn't affect domain rules.

### ISSUE THREAD

**Issue by lixiaoyan** [biomejs/biome#6033]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         xterm-ghostty
  JS_RUNTIME_VERSION:           v24.0.2
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.4.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

1. Clone https://github.com/lixiaoyan/biome-recommended-domains-repro
2. `npm i && npx biome check`
3. Biome throws a lint error: `lint/correctness/useHookAtTopLevel`

**biome.json**:
```json
{
	"linter": {
		"rules": {
			"recommended": false
		}
	}
}
```

**test.js**:
```js
const Comp = () => {
	if (true) {
		useState();
	}
};
```

### Expected result

It should not throw an error because there are no linter rules configured.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
However, I suppose when `recommended` is set to `false`, even domains should adhere regardless of the dependencies
